### PR TITLE
improve ebnf syntax

### LINKED ISSUE #123: Improve EBNF Syntax
Our current EBNF matches the Grammar YAML, not actual EBNF syntax. For the spec, we need to use valid/conformant EBNF, so that it is readable and usable by readers. This issue tracks changing the EBNF generator to use [standard syntax](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form).

Note that this will not change the Grammar schema itself, as the additional nodes are needed for parser optimizations.

### PR DESCRIPTION
**Pull request by OmarTawfik**
- update ebnf grammar to latest syntax
- fix a bunch of precedence issues when generating sub expressions
- break really long lines (operators, keywords)
- indicate associativity via comments

closes #123

### CODE REVIEW

**Review by AntonyBlakey — CHANGES_REQUESTED** [review]
Parens don't work for associativity because either recursive branch could be taken. Comments are the best option.

**Comment by OmarTawfik** [comment]
f2f: Will remove parens and switch to use comments for now.

File: crates/codegen/ebnf/src/serialization.rs [hunk 1]
```diff
@@ -80,46 +105,43 @@ impl<'grammar> EbnfSerializer<'grammar> {
...
-                self.buffer.push_str("]");
-            }
-            EbnfNode::Parenthesis { body } => {
-                self.buffer.push_str("(");
-                self.serialize_node(*body);
-                self.buffer.push_str(")");
+                self.serialize_child_node(node, body);
+                self.buffer.push_str("?");
             }
             EbnfNode::Range { from, to } => {
```
Comment (OmarTawfik) [inline crates/codegen/ebnf/src/serialization.rs:124]: `[a-z]`

File: crates/codegen/ebnf/src/precedence_parser.rs [hunk 2]
```diff
@@ -4,39 +4,49 @@ use crate::{nodes::EbnfNode, serialization::GenerateEbnf};
 
 impl GenerateEbnf for PrecedenceParserRef {
     fn generate_ebnf(&self) -> EbnfNode {
-        let mut alternatives = vec![];
+        let mut choices = vec![];
 
         for operator in &self.definition.operators {
```
Comment (OmarTawfik) [inline crates/codegen/ebnf/src/precedence_parser.rs:9]: f2f: Will remove parens and switch to use comments to indicate associativity.

**Comment by OmarTawfik** [comment]
Using comments now to indicate associativity.

**Review by AntonyBlakey — APPROVED** [review]

**Pull request merged**